keanu-project/src/test/java/io/improbable/keanu/algorithms/variational/KDEApproximationTest.java

```diff
@@ -34,10 +33,13 @@
 
     public DoubleVertexSamples generateGaussianSamples(double mu, double sigma, int nSamples) {
         DoubleVertex gaussian = new GaussianVertex(mu, sigma);
-        BayesianNetwork network = new BayesianNetwork(gaussian.getConnectedGraph());
```

Comment:
💯

Reply:
On my machine I think it went from 1:50 to 49 seconds - so quite substantially faster.